### PR TITLE
Ios bug

### PR DESCRIPTION
**Pull request by Lexseal**
### Features Added/Issues Addressed
fixed ios filter issue

### Brief Description of Solution
Turns out ios adds a white space at the end of each label, causing the set not to contain the preference. Did a trim each time we read.

### Checklist
- [ ] Added new tests (You know the code best to make new tests)
- [x] Resolved merge conflicts (Prefer merge over rebase)
- [x] Status check passed (Linted and tested)
- [ ] Reviewed (At least one reviewer's approval)

### Caveat/Special Instructions for Team
None

### CODE REVIEW

File: source/assets/components/ExpandedRecipeCard.js [hunk 1]
```diff
@@ -141,20 +141,20 @@ class RecipeCard extends HTMLElement {
...
-        if (setLevel > 0.75) 
-            numLevel = 4;
-        else if (setLevel > 0.50) 
-            numLevel = 3;
-        else if (setLevel > 0.25) 
-            numLevel = 2;
-        else 
-            numLevel = 1;
-        level.textContent = numLevel;
+      const setLevel = data.difficulty;
```
Comment (Lexseal) [inline source/assets/components/ExpandedRecipeCard.js:144]: ignore this (lint)

File: source/assets/scripts/backend.js [hunk 2]
```diff
@@ -410,8 +410,10 @@ export function filter_intolerance(recipe_hashes, intolerances) {
   
   return recipe_hashes.filter(recipe_hash => {
     const recipe_tags = new Set(get_recipe(recipe_hash).intolerances);
-    for (let intolerace of intolerances)
-      if (!recipe_tags.has(intolerace)) return false;
+    for (let intolerace of intolerances)
```
Comment (Lexseal) [inline source/assets/scripts/backend.js:413]: ignore this (lint)

File: source/assets/scripts/foodie.js [hunk 3]
```diff
@@ -2,11 +2,11 @@
 
 import * as backend from './backend.js';
 
-if (localStorage.getItem('%not_first_visit')) {
+if (localStorage.getItem('%not_first_visit'))
```
Comment (Lexseal) [inline source/assets/scripts/foodie.js:5]: ignore this (lint)

File: source/assets/scripts/index.js [hunk 4]
```diff
@@ -3,11 +3,11 @@
 
 import * as backend from './backend.js';
 
-if (localStorage.getItem('%not_first_visit')) {
+if (localStorage.getItem('%not_first_visit'))
```
Comment (Lexseal) [inline source/assets/scripts/index.js:6]: ignore this (lint)

File: source/assets/scripts/onBoardingPage.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ function begin() {
     for(let i = 0; i < leftCkbox.length; i++){
       let ingredientBox = leftCkbox[i].getElementsByTagName('input')[0];
       if(ingredientBox.checked){
-        let ingredientText = leftCkbox[i].innerText;
+        let ingredientText = leftCkbox[i].innerText.trim();
```
Comment (Lexseal) [inline source/assets/scripts/onBoardingPage.js:30]: Added trim here

**Review by Xubangbangg — APPROVED** [review]
Thanks about fixing the bug! It looks good for me

**Pull request merged**